src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processhearingrecordingrequest/ProcessHearingRecordingRequestAboutToSubmitHandler.java

```diff
@@ -69,43 +72,52 @@ private void processHearingRecordings(SscsCaseData sscsCaseData,
                                           ProcessHearingRecordingRequest processHearingRecordingRequest,
                                           PartyItemList partyItemList) {
 
-
-        SscsHearingRecordingCaseData sscsHearingRecordingCaseData =
-                sscsCaseData.getSscsHearingRecordingCaseData();
         ProcessHearingRecordingRequestDetails processHearingRecordingRequestValue =
                 processHearingRecordingRequest.getValue();
 
         String status = null;
         switch (partyItemList) {
             case DWP:
                 DynamicList dwpRequestStatus = processHearingRecordingRequestValue.getDwp();
-                if (dwpRequestStatus != null) {
+                if (nonNull(dwpRequestStatus) && nonNull(dwpRequestStatus.getValue())
+                        && nonNull(dwpRequestStatus.getValue().getCode())) {
                     status = dwpRequestStatus.getValue().getCode();
                     break;
                 } else {
                     return;
                 }
-                //TODO:Add unit tests for this, should be part of SSCS-9239
             case APPELLANT:
                 DynamicList appellantRequestStatus = processHearingRecordingRequestValue.getAppellant();
-                if (appellantRequestStatus != null) {
+                if (nonNull(appellantRequestStatus) && nonNull(appellantRequestStatus.getValue())
+                        && nonNull(appellantRequestStatus.getValue().getCode())) {
                     status = appellantRequestStatus.getValue().getCode();
                     break;
                 } else {
                     return;
                 }
             case JOINT_PARTY:
                 DynamicList jointPartyStatus = processHearingRecordingRequestValue.getJointParty();
-                if (jointPartyStatus != null) {
+                if (nonNull(jointPartyStatus) && nonNull(jointPartyStatus.getValue())
+                        && nonNull(jointPartyStatus.getValue().getCode())) {
                     status = jointPartyStatus.getValue().getCode();
                     break;
                 } else {
                     return;
                 }
+            case REPRESENTATIVE:
+                DynamicList representativeStatus = processHearingRecordingRequestValue.getRep();
+                if (nonNull(representativeStatus) && nonNull(representativeStatus.getValue())
+                        && nonNull(representativeStatus.getValue().getCode())) {
+                    status = representativeStatus.getValue().getCode();
+                    break;
+                } else {
+                    return;
+                }
```

Comment:
Is there a way of cleaning all this up? Feels like some of it can be combined into a cleaner method rather than having repeated logic for each party...

Reply:
Might have done something similar in the AboutToStart, so it's worth adding it to the common service method

Reply:
I refactored it a bit. Please, have a look